units/systemd-quotacheck-root.service.in

```diff
@@ -0,0 +1,24 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=File System Quota Check
+Documentation=man:systemd-quotacheck.service(8)
+
+ConditionPathExists={{QUOTACHECK}}
+
+DefaultDependencies=no
+After=initrd-switch-root.target
```

Comment:
why this dep here? do you envision the quota stuff to run inside the initrd? i had assumed it's run only after the transition.

In fact, all four services should probably have ConditionPathExists=!/etc/initrd-release so that they become NOPs if used in the initrd...

Reply:
Actually, quotacheck needs to run when the filesystem is rw, but before anything else is writing to it.
Now, I'm using:

After=systemd-remount-fs.service
Before=systemd-journal-flush.service shutdown.target

hope this is addressed correctly like that.